src/main/java/org/isf/utils/jobjects/PaginatedTableDecorator.java

```diff
@@ -0,0 +1,179 @@
+package org.isf.utils.jobjects;
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.GridLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.swing.Box;
+import javax.swing.ButtonGroup;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.JToggleButton;
+import javax.swing.SwingConstants;
+import javax.swing.event.RowSorterEvent;
+import javax.swing.event.RowSorterListener;
+import javax.swing.event.TableModelEvent;
+import javax.swing.table.TableModel;
+
+public class PaginatedTableDecorator<T> {
+    private JTable table;
+    private PaginationDataProvider<T> dataProvider;
+    private int[] pageSizes;
+    private JPanel contentPanel;
+    private int currentPageSize;
+    private int currentPage = 1;
+    private JPanel pageLinkPanel;
+    private PageableTableModel objectTableModel;
+    private static final int MaxPagingCompToShow = 9;
+    private static final String Ellipses = "...";
+
+    private PaginatedTableDecorator(JTable table, PaginationDataProvider<T> dataProvider,
+                                    int[] pageSizes, int defaultPageSize) {
+        this.table = table;
+        this.dataProvider = dataProvider;
+        this.pageSizes = pageSizes;
+        this.currentPageSize = defaultPageSize;
+    }
+
+    public static <T> PaginatedTableDecorator<T> decorate(JTable table,
+                                                          PaginationDataProvider<T> dataProvider,
+                                                          int[] pageSizes, int defaultPageSize) {
+        PaginatedTableDecorator<T> decorator = new PaginatedTableDecorator<>(table, dataProvider,
+                pageSizes, defaultPageSize);
+        decorator.init();
+        return decorator;
+    }
+
+    public JPanel getContentPanel() {
+        return contentPanel;
+    }
+
+    private void init() {
+        initDataModel();
+        initPaginationComponents();
+        initListeners();
+        paginate();
+    }
+
+    private void initListeners() {
+        objectTableModel.addTableModelListener(this::refreshPageButtonPanel);
+        if (table.getRowSorter() != null) {
+            table.getRowSorter().addRowSorterListener(new RowSorterListener() {
+                @Override
+                public void sorterChanged(RowSorterEvent e) {
+                    if(e.getType()== RowSorterEvent.Type.SORT_ORDER_CHANGED) {
```

Comment:
missing space between `if` and `(`; also missing space before `==`

Reply:
Done, thanks!